test/e2e/specs/all-features-confirmation/record/presentation.spec.js

```diff
@@ -163,4 +170,92 @@ describe('View existing record,', function() {
         });
     });
 
+    describe("For side panel table of contents in Record App", function() {
+
+        beforeAll(function() {
+            var url = browser.params.url + "/record/#" + browser.params.catalogId + "/" + testParams.sidePanelTest.schemaName + ":" + testParams.sidePanelTest.tableName +  "/id=" + testParams.sidePanelTest.id;
+            browser.get(url);
+            recSidePan = chaisePage.recordPage.getElementById('recordSidePan-5');
+            fiddlerBtn = element(by.className('sidePanFiddler')).element(by.tagName('i')),
+            chaisePage.waitForElement(fiddlerBtn);
+        });
+        it('Table of contents should be collapsed by default', function(){
+            var recordSidePan = chaisePage.recordPage.getElementById('record-side-pan');
+            recordSidePan.isDisplayed().then(function (bool) {
+                expect(bool).toBeFalsy();
+            });
+        });
+
+        it('Table of contents should match all related table', function(){
+            var  allPan = element.all(by.xpath("//div[contains(@id,'recordSidePan')]"));
+            chaisePage.waitForElement(fiddlerBtn).then(function(){
+              return fiddlerBtn.click();
+            }).then(function(){
+              expect(allPan.count()).toBe(testParams.sidePanelTest.tocCount, "Table of contents count did not match!");
+            }).catch( function(err) {
+               console.log(err);
+           });
+        });
+
+        it('On click of Related table name in TOC, page should move to the contents and open the table details', function(){
+            var rtTableHeading = chaisePage.recordPage.getRelatedTableHeading(testParams.sidePanelTest.tableToShow),
+                lastRow = element(by.id('rt-heading-Categories_5')).all(by.tagName('tr')).get(5);
+
+            recSidePan.getAttribute('class').then(function(className) {
+                return recSidePan.click();
+            }).then (function (){
+              // related table should be visible
+                rtTableHeading.isDisplayed().then(function (bool) {
+                  expect(bool).toBeTruthy();
+                });
+                chaisePage.waitForElement(lastRow);
+                return rtTableHeading.getAttribute("class");
+            }).then (function(className) {
+                expect(className).toContain("panel-open", "Related table panel is not open when clicked through TOC.");
+            }).catch( function(err) {
+                console.log(err);
+            });
+        });
+
+        it('Record count should match for pan and related table content', function(){
+            var rtRow = browser.executeScript("return $('#rt-Categories_5 tbody tr')"),
+                rtPanRecordCount = element(by.id('recordSidePan-heading-5')).all(by.tagName('span')).get(1),
```

Comment:
This should be changed to fetch the full title of each heading in the side panel.